cartographer_grpc/framework/rpc.cc

```diff
@@ -77,7 +78,7 @@ void Rpc::RequestNextMethodInvocation() {
   // Ask gRPC to notify us when the connection terminates.
   SetRpcEventState(Event::DONE, true);
   server_context_.AsyncNotifyWhenDone(
-      new RpcEvent{Event::DONE, weak_ptr_factory_(this)});
+      new RpcEvent{Event::DONE, weak_ptr_factory_(this), event_queue_});
```

Comment:
Isn't the event queue already clear from the rpc, is it better to avoid passing it?

Reply:
Ok removed the EventQueue* member from RpcEvent. The only drawback is that we need to upgrade the weak_ptr to a shared_ptr twice. But agreed let's deal with that once it becomes a bottleneck.